### PR TITLE
HW-2 was done

### PR DESCRIPTION
**Pull request by dmytrovb02**

### CODE REVIEW

File: src/main/java/com/bookstore/config/MapperConfig.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.bookstore.config;
+
+import org.mapstruct.InjectionStrategy;
+import org.mapstruct.NullValueCheckStrategy;
+
+@org.mapstruct.MapperConfig(
+        componentModel = "spring",
+        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+        implementationName = "<PACKAGE_NAME.impl"
```
Comment (Rommelua) [inline src/main/java/com/bookstore/config/MapperConfig.java]:
```suggestion
        implementationName = "<PACKAGE_NAME>.impl"
```

File: src/main/java/com/bookstore/exception/message/ExceptionMessage.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.bookstore.exception.message;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@Getter
+@AllArgsConstructor
+public enum ExceptionMessage {
```
Comment (Rommelua) [inline src/main/java/com/bookstore/exception/message/ExceptionMessage.java]: delete it. In a real app you can have 1000 entities. So, you will have 1000 enum values only for "not found"

File: src/main/java/com/bookstore/controller/BookController.java [hunk 3]
```diff
@@ -1,7 +1,39 @@
...
+    }
+
+    @GetMapping("/{id}")
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.findById(id);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public void createBook(@RequestBody BookRequestDto bookRequestDto) {
```
Comment (Rommelua) [inline src/main/java/com/bookstore/controller/BookController.java]:
```suggestion
    public BookResponseDto createBook(@RequestBody BookRequestDto bookRequestDto) {
```

File: pom.xml [hunk 4]
```diff
@@ -46,20 +61,46 @@
 			<scope>test</scope>
 		</dependency>
 		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-validation</artifactId>
+			<groupId>org.hibernate.validator</groupId>
+			<artifactId>hibernate-validator</artifactId>
+			<version>8.0.1.Final</version>
```
Comment (sokimaaa) [inline pom.xml]: Use starter. 
```suggestion
			<groupId>org.hibernate.validator</groupId>
			<artifactId>hibernate-validator</artifactId>
			<version>8.0.1.Final</version>
```

File: src/main/java/com/bookstore/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -12,14 +16,26 @@
...
+    public List<BookResponseDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::mapToDto)
+                .toList();
+    }
+
+    @Override
+    public BookResponseDto findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException(String.format("Book with id: %s not found", id)));
```
Comment (sokimaaa) [inline src/main/java/com/bookstore/service/impl/BookServiceImpl.java]: use Optional#map to inline code
```suggestion
        Book book = bookRepository.findById(id).orElseThrow(
                () -> new EntityNotFoundException(String.format("Book with id: %s not found", id)));
```

**Pull request merged**